src/telemetry.ts

```diff
@@ -1626,23 +1823,119 @@ export class IEventNamePropertyMapping {
          * nativeNotebookToolbar - Invoked from Native notebook toolbar.
          */
         commandSource: SelectJupyterUriCommandSource;
-    }> = { owner: 'IanMatthewHuff' } as any;
-    [Telemetry.SetJupyterURIToLocal]: TelemetryEventInfo<DurationMeasurement> = { owner: 'IanMatthewHuff' } as any;
+    }> = {
+        owner: 'IanMatthewHuff',
+        feature: ['KernelPicker'],
+        source: 'N/A',
+        properties: {
+            commandSource: {
+                classification: 'SystemMetaData',
+                purpose: 'FeatureInsight'
+            }
+        }
+    };
+    /**
+     * Jupyter URI was set to local.
+     */
+    [Telemetry.SetJupyterURIToLocal]: TelemetryEventInfo<DurationMeasurement> = {
+        owner: 'IanMatthewHuff',
+        feature: ['KernelPicker'],
+        source: 'N/A',
+        measures: commonClassificationForDurationProperties
+    };
+    /**
+     * Jupyter URI was valid and set to a remote setting.
+     */
     [Telemetry.SetJupyterURIToUserSpecified]: TelemetryEventInfo<{
+        /*
+         * Was the URI set to an Azure uri.
+         */
         azure: boolean;
-    }> = { owner: 'IanMatthewHuff' } as any;
+    }> = {
+        owner: 'IanMatthewHuff',
+        feature: ['KernelPicker'],
+        source: 'N/A',
+        properties: {
+            azure: {
+                classification: 'SystemMetaData',
+                purpose: 'FeatureInsight'
+            }
+        }
+    };
     [Telemetry.ShiftEnterBannerShown]: TelemetryEventInfo<never | undefined> = {
         owner: 'amunger'
     } as any;
-    [Telemetry.StartShowDataViewer]: TelemetryEventInfo<DurationMeasurement> = { owner: 'IanMatthewHuff' } as any;
-    [Telemetry.ShowDataViewer]: TelemetryEventInfo<{ rows: number | undefined; columns: number | undefined }> = {
-        owner: 'IanMatthewHuff'
-    } as any;
-    [Telemetry.FailedShowDataViewer]: TelemetryEventInfo<never | undefined> = { owner: 'IanMatthewHuff' } as any;
+    /**
+     * User requested to open the data frame viewer.
+     */
+    [Telemetry.StartShowDataViewer]: TelemetryEventInfo<DurationMeasurement> = {
+        owner: 'IanMatthewHuff',
+        feature: ['DataFrameViewer'],
+        source: 'User Action',
+        measures: commonClassificationForDurationProperties
+    };
+    /**
+     * Request was made to show the data viewer with specific data frame info.
+     */
+    [Telemetry.ShowDataViewer]: TelemetryEventInfo<{
+        /**
+         * Count of rows in the target data frame.
+         */
+        rows: number | undefined;
+        /**
+         * Count of columns in the target data frame.
+         */
+        columns: number | undefined;
+    }> = {
+        owner: 'IanMatthewHuff',
+        source: 'N/A',
+        feature: ['DataFrameViewer'],
+        properties: {
+            rows: {
+                classification: 'PublicNonPersonalData',
```

Comment:
@DonJayamanne I went back and forth on it a bit. From my reading I thought that it might be PublicNonPersonal as it's not under our control, it's up to the customer who had made the size of the data frame. But it's non personal.

Reading the categories more, you might be right? Feel it's a bit hard to tell.